File: MyshopFrontend/src/app/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Product } from './model/product';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private baseUrl = 'http://localhost:1990/';

  constructor(private http:HttpClient) { }

  create(products:Product){
    return this.http.post(`${this.baseUrl}`+'product',products);
  }
  
  getAll(){
    return this.http.get(this.baseUrl+'product');
  }

  get(productId){
    return this.http.get(this.baseUrl+'product/'+productId);
  }

  update(productId:number,product:Product){
    return this.http.post<Product>(this.baseUrl+'updateProduct/'+productId,product);
   
  }

  delete(productId){
    return this.http.delete(this.baseUrl+'product/'+productId);
  }
}

File: MyShopBackend/ProductService/src/main/java/lk/nasi/ProductService/service/ProductService.java
package lk.nasi.ProductService.service;

import java.util.List;
import java.util.Optional;

import lk.nasi.ProductService.model.Product;

public interface ProductService {

	Product createOrUpdate(Product products);
		  
	List<Product> getAll();

	Optional<Product> get(int productId);

	
	void delete(int productId);
}
